keanu-python/keanu/algorithm/sampling.py

```diff
@@ -43,17 +53,46 @@ def sample(net: BayesNet,
 def generate_samples(net: BayesNet,
                      sample_from: Iterable[Vertex],
                      algo: str = 'metropolis',
+                     proposal_distribution: str = None,
+                     proposal_distribution_sigma: numpy_types = None,
+                     proposal_listeners: List[proposal_listener_types] = [],
                      drop: int = 0,
                      down_sample_interval: int = 1) -> sample_generator_types:
-    vertices_unwrapped = k.to_java_object_list(sample_from)
 
-    sample_iterator = algorithms[algo].withDefaultConfig().generatePosteriorSamples(net.unwrap(), vertices_unwrapped)
-    sample_iterator = sample_iterator.dropCount(drop).downSampleInterval(down_sample_interval)
-    sample_iterator = sample_iterator.stream().iterator()
+    sampling_algorithm: JavaObject = build_sampling_algorithm(algo, proposal_distribution, proposal_distribution_sigma,
+                                                              proposal_listeners)
+
+    vertices_unwrapped: JavaList = k.to_java_object_list(sample_from)
+
+    samples: JavaObject = sampling_algorithm.generatePosteriorSamples(net.unwrap(), vertices_unwrapped)
+    samples = samples.dropCount(drop).downSampleInterval(down_sample_interval)
+    sample_iterator: JavaObject = samples.stream().iterator()
 
     return _samples_generator(sample_iterator, vertices_unwrapped)
 
 
+def build_sampling_algorithm(algo, proposal_distribution: Optional[str],
+                             proposal_distribution_sigma: Optional[numpy_types],
+                             proposal_listeners: List[proposal_listener_types]):
+    if algo != "metropolis":
```

Comment:
Not really, because the mechanism on the Java side is quite different: you construct a `ProposalDistribution` object and pass it to the `MetropolisHastingsBuilder`. There'd be nowhere to put a `NotImplementedError`.